include/envoy/config/incremental_subscription.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/common/pure.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Config {
+
+template <class ResourceType> class IncrementalSubscriptionCallbacks {
```

Comment:
I think you should get a WiP e2e in place before we start merging; there's a few things here that I think will change substantially, e.g. resource versioning, so would be good to see a full WIP PR before deciding on interfaces. The WIP doesn't need to be fully complete for error handling or tests (maybe just a set of xDS updates in an integration test will do), that should let us know we are on the right track.

Reply:
Ok, will do.